xlators/protocol/server/src/server-rpc-fops_v2.c

```diff
@@ -4360,6 +4419,17 @@ server4_0_setxattr(rpcsvc_request_t *req)
         goto out;
     }
 
+    /* Let the namespace setting can happen only from special mounts, this
+       should prevent all mounts creating fake namespace. */
+    if ((frame->root->pid >= 0) &&
+        dict_get_sizen(state->dict, GF_NAMESPACE_KEY)) {
+        gf_smsg("server", GF_LOG_ERROR, 0, PS_MSG_SETXATTR_INFO, "path=%s",
+                state->loc.path, "key=%s", GF_NAMESPACE_KEY, NULL);
+        ret = -1;
```

Comment:
remove 'ret = -1` line, it is confusing, I had to check what SERVER_REQ_SET_ERROR() is doing to ret again.

Reply:
done